castoredc_api/client/castoredc_api_client.py

```diff
@@ -684,14 +812,28 @@ def create_survey_package_instance(
         }
         return self.sync_post(url, body)
 
-    def patch_survey_package_instance(self, survey_package_instance_id, status):
+    def lock_unlock_survey_package_instance(self, survey_package_instance_id, status):
         """Lock/unlock survey package."""
-        url = self.study_url + "/surveypackageinstance/" + survey_package_instance_id
+        url = self.study_url + f"/surveypackageinstance/{survey_package_instance_id}"
         body = {
             "locked": status,
         }
         return self.sync_patch(url, body)
 
+    def update_start_time_survey_package_instance(
+        self, record_id: str, survey_package_instance_id: str, datetime: str
```

Comment:
I'm not familiar with EDC's API (I just handle infra stuff :stuck_out_tongue:) so maybe taking a string for datetime makes sense for our API, however this looks a little sketchy to me.

When dealing with dates, I'd expect Python datetime objects as input to reduce the chances of this function accepting bogus inputs.

If you stick with strings, I'd at least validate (with client-side code) that the input you got really matches the `yyyy-mm-dd hh:mm:ss` format, instead of blindly trusting our API to reject invalid input.

Reply:
I'll add datetime object support for these type of functions and validation of string format.